src/inputs/TreeSelectField/TreeSelectField.tsx

```diff
@@ -94,8 +94,16 @@ export function TreeSelectField<O, V extends Value>(
   } = props;
 
   const [collapsedKeys, setCollapsedKeys] = useState<Key[]>(
-    Array.isArray(options) && defaultCollapsed ? options.map((o) => getOptionValue(o)) : [],
+    !Array.isArray(options)
+      ? []
+      : defaultCollapsed
+      ? options.map((o) => getOptionValue(o))
+      : options
+          .flatMap(flattenOptions)
+          .filter((o) => o.defaultCollapsed)
+          .map((o) => getOptionValue(o)),
```

Comment:
issue: usually if we haven't load the options, default to an empty array, this will provoke the `option.defaultCollapsed` not be applied
```
treeFilter({
        label: "Locations",
        filterBy: "all",
        getOptionLabel: (o) => o.name,
        getOptionValue: (o) => o.id,
        options: locationOptions ?? [],
      }),
```